scripts/corelli/calibration/utils.py

```diff
@@ -91,13 +93,67 @@ def load_banks(filename: str, bank_selection: str, output_workspace: str) -> Wor
-                               shadow_width: float = 4, fit_domain: float = 7) -> TableWorkspace:
+def calculate_peak_y_table(peak_table: InputTable,
+                           parameters_table_group: WorkspaceGroupTypes,
+                           output_workspace: str = 'PeakYTable') -> TableWorkspace:
+    r"""
+    Evaluate the Y-coordinate of each wire shadow given the estimated pixel positions of the
+    shadows and the polynomial function translating pixel positions to y-coordinate
+
+    :param peak_table:
+    :param parameters_table_group:
+    :param output_workspace:
+
+    :return: TableWorkspace
+    """
+    # Number of rows of peak_table must be the number of tables within parameters_table_group
+    # Create empty PeakYTable, see tube_calib.getCalibration
+    # Iterate over the rows of peak_table, and tables of parameters_table_group
+    #   Extract peak positions in pixel units
+    #   Extract polynomial coefficients, see tube_calib.correct_tube_to_ideal_tube
+    #   Evaluate the Y-coordinate for the peak positions, see tube_calib.correct_tube_to_ideal_tube
+    #   Append row to PeakYTable, see tube_calib.getCalibration
+    # Return the handle to PeakYTable
+    peak_workspace, parameters_workspace = mtd[str(peak_table)], mtd[str(parameters_table_group)]
+    error_message = 'number of rows in peak_table different than number of tables in parameters_table_group'
+    assert peak_workspace.rowCount() == parameters_workspace.getNumberOfEntries(), error_message
+
+    # Create empty peak_y_table with same column names as those of peak_table
+    peak_y_workspace = CreateEmptyTableWorkspace(OutputWorkspace=output_workspace)
+    for column_type, column_name in zip(peak_workspace.columnTypes(), peak_workspace.getColumnNames()):
+        peak_y_workspace.addColumn(type=column_type, name=column_name)
+    tube_names = peak_workspace.column(0)  # e.g., ['CORELLI/A row/bank10/sixteenpack/tube1',...]
+
+    # calculate the vertical positions for each tube, and append to peak_y_workspace
+    for tube_index, (row, table) in enumerate(zip(peak_workspace, parameters_workspace)):
+        peak_pixel_positions = list(row.values())[1:]  # peaks in pixel units for the current tube
+        coefficients = table.column(1)[:-1]  # A0, A1, A2 coefficients
+        peak_vertical_positions = polyval(peak_pixel_positions, coefficients)  # peaks along vertical axes
+        peak_y_workspace.addRow([tube_names[tube_index]] + peak_vertical_positions.tolist())
+
+    return peak_y_workspace
+
+
+def calibrate_tube(workspace: WorkspaceTypes,
+                   tube_name: str,
+                   output_peak_table: str = 'PeakTable',
+                   output_parameters_table: str = 'ParametersTable',
+                   output_peak_y_table: str = 'PeakYTable',
+                   shadow_height: float = 1000,
+                   shadow_width: float = 4,
+                   fit_domain: float = 7) -> TableWorkspace:
```

Comment:
Just add comment for hardcoded values, unless they are obvious.

Reply:
I introduce an explanation of the hardcoded values for `shadow_height`, `shadow_width`, and `fit_domain` in the docstring of the function